File: server/src/main/java/org/tcpx/sharine/vo/VideoVO.java
package org.tcpx.sharine.vo;

import jakarta.persistence.Column;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.tcpx.sharine.entity.Video;
import org.tcpx.sharine.utils.BeanCopyUtils;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class VideoVO {
    Long id;
    // 不一致字段 userId -> author
    UserDetailVO author;

    String title;

    String url;

    String coverUrl;
    // 不存在字段 通过 VideoService 获取
    Integer favoriteCount;
    // 不存在字段 通过 VideoService 获取
    Integer bookmarkCount;
    // 不存在字段 通过 VideoCategoryService 获取
    CategoryVO category;

    String content;

    Long viewCount;

    Long updateTime;

    Long createTime;

    @SuppressWarnings("请使用相关 Service 类的 build 方法进行构造，否则会缺失参数")
    public static VideoVO of(Video video) {
        return BeanCopyUtils.copyObject(video, VideoVO.class);
    }
}

File: web/src/models/VideoVO.ts
import {VideoType} from "./VideoType";
import {VideoComment} from "./VideoComment";
import {User} from "@/models/User";
import {Category} from "@/models/Category";

export class VideoVO {
    id: number = 0
    author: User = new User()
    category: Category = new Category()
    title: string = ""
    content: string = ""
    url: string = ""
    coverUrl: string = ""
    type: VideoType = VideoType.UNKNOWN
    comments: Array<VideoComment> = new Array<VideoComment>()
    bookmarkCount: number = 0
    favoriteCount: number = 0
    viewCount: number = 0
    createTime: Date = new Date()
    updateTime: Date = new Date()
}